### PR TITLE
Module lint and clean up. Implemented fixes from testing on Azure

### PR DESCRIPTION
**Pull request by wilr**
Deployed to Azure MSSQL and confirmed dev/build

### CODE REVIEW

File: src/SQLServerConnector.php [hunk 1]
```diff
@@ -143,18 +161,34 @@ public function preparedQuery($sql, $parameters, $errorLevel = E_USER_ERROR)
 
         // Check for error
         if (!$handle) {
+            $error = $this->getLastError();
+
+            if (preg_match("/Cannot insert explicit value for identity column in table '(.*)'/", $error, $matches)) {
```
Comment (dhensby) [inline src/SQLServerConnector.php:166]: This seems risky - if the database has `SET_IDENTITY_INSERT` off then we shouldn't be turning it on to force the query. The application should assume the that DB Server is correctly configured, rather than running these implicit queries.

File: src/SQLServerQuery.php [hunk 2]
```diff
@@ -43,27 +47,10 @@ public function __destruct()
         }
     }
 
-    public function getIterator()
```
Comment (dhensby) [inline src/SQLServerQuery.php:46]: Why has this generator been replaced with the old style API? https://github.com/silverstripe/silverstripe-mssql/pull/49

File: src/MSSQLSchemaManager.php [hunk 3]
```diff
@@ -135,27 +134,6 @@ public function getPrimaryKey($tableName)
         return $indexName;
     }
 
-    /**
-     * Gets the identity column of a table
-     *
-     * @param string $tableName
-     * @return string|null
-     */
-    public function getIdentityColumn($tableName)
```
Comment (dhensby) [inline src/MSSQLSchemaManager.php:144]: Why has this function been removed?

**Review by dhensby — CHANGES_REQUESTED** [review]
Amazing work, thanks for doing this.

Only a couple of questions.

I also think that this should probably constitute a new major release, especially if some of the APIs are being removed - that means bumping the master alias in composer.json.

File: src/SQLServerConnector.php (hunk 1 above)
Comment (wilr) [reply src/SQLServerConnector.php:166]: I believe that IDENTITY INSERT can only be overridden one table at a time, per connection and there isn't a global method configuration option for it?

File: src/SQLServerQuery.php (hunk 2 above)
Comment (wilr) [reply src/SQLServerQuery.php:46]: @dhensby I take it `getIterator` is for SS5 not SS4 though? I was trying to get compatibility with SS 4.6

File: src/MSSQLSchemaManager.php (hunk 3 above)
Comment (wilr) [reply src/MSSQLSchemaManager.php:144]: As far as I could see it was no longer used anywhere?

File: src/SQLServerConnector.php (hunk 1 above)
Comment (dhensby) [reply src/SQLServerConnector.php:166]: hmm - inserting with ID values is pretty central to SS - it's required for almost all our ORM inserts where there's MTI in play. Having logic that runs a query, checks the error, runs another query, re-runs the original and then runs a fourth final query is going to have a massive impact on performance.

Not only that, `SET_IDENTITY_INSERT` requires `ALTER` permissions on the table, so anyone with appropriately locked down permissions for their app won't even be able to run the query anyway.

However, it does seem to be somewhat impossible to have an autoincrement value in MSSQL without using the IDENTITY type. This just provides more reason why we shouldn't rely on the DB layer to produce PK values in SS.

In conclusion, it looks like we have pretty much no choice about this and we should probably accept that MSSQL is not a good DBMS to be using with SS. :(

File: src/SQLServerQuery.php (hunk 2 above)
Comment (dhensby) [reply src/SQLServerQuery.php:46]: Hmm - ok, then we need to do some branch management on this repo, it seems.
